packages/filesystem/src/browser/file-dialog/file-dialog.ts

```diff
@@ -218,3 +275,86 @@ export class FileDialog extends AbstractDialog<MaybeArray<FileStatNode>> {
     }
 
 }
+
+@injectable()
+export class SaveFileDialog extends FileDialog<string> {
```

Comment:
Why not `URI`?

Reply:
It's not a problem to rework it on URI. Which URI you prefer?
URI from packages/core/src/common/uri.ts or Uri from 'vscode-uri' ?

Reply:
It is not allowed to expose `vscode-uri` to developers, it is an implementation detail of Theia `URI`.

Reply:
Reworked. Thanks!